From 5d3c35c19083053128ec2c48e6393c2a9803d57e Mon Sep 17 00:00:00 2001
From: Dorian Clay <dorian@dorianclay.com>
Date: Sun, 30 Jan 2022 11:50:09 -0800
Subject: [PATCH] hwdb: add support for Surface Laptop 2 & 3 (#22303)

(cherry picked from commit 156d01b9ca4cb6d7b3877a3db7d20ed600dc40c0)

Related: #2087778
---
 hwdb.d/60-evdev.hwdb | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/hwdb.d/60-evdev.hwdb b/hwdb.d/60-evdev.hwdb
index f896dde357..9fcb4a3ddf 100644
--- a/hwdb.d/60-evdev.hwdb
+++ b/hwdb.d/60-evdev.hwdb
@@ -591,6 +591,24 @@ evdev:name:MSFT0001:02 04F3:304B Touchpad:dmi:*svnLENOVO:*pvrLenovoLegionY9000X2
  EVDEV_ABS_35=::31
  EVDEV_ABS_36=::30
 
+#########################################
+# Microsoft
+#########################################
+
+# Surface Laptop 2 (13")
+evdev:name:Microsoft Surface 045E:0933 Touchpad:dmi:*svnMicrosoftCorporation:*pnSurfaceLaptop2**
+ EVDEV_ABS_00=::38
+ EVDEV_ABS_01=::38
+ EVDEV_ABS_35=::38
+ EVDEV_ABS_36=::38
+
+# Surface Laptop 3 (15")
+evdev:name:Microsoft Surface 045E:09AF Touchpad:dmi:*svnMicrosoftCorporation:*pnSurfaceLaptop3**
+ EVDEV_ABS_00=::39
+ EVDEV_ABS_01=::37
+ EVDEV_ABS_35=::39
+ EVDEV_ABS_36=::37
+
 #########################################
 # NEWYES
 #########################################
